pyretailscience/analysis/revenue_tree.py

```diff
@@ -38,6 +38,180 @@
+    df_cols = df.columns
+
+    if cols.agg_unit_qty in df_cols:
+        df[cols.calc_units_per_trans] = df[cols.agg_unit_qty] / df[cols.agg_transaction_id]
+        df[cols.calc_price_per_unit] = df[cols.agg_unit_spend] / df[cols.agg_unit_qty]
+
+    df[cols.calc_spend_per_cust] = df[cols.agg_unit_spend] / df[cols.agg_customer_id]
+    df[cols.calc_spend_per_trans] = df[cols.agg_unit_spend] / df[cols.agg_transaction_id]
+    df[cols.calc_trans_per_cust] = df[cols.agg_transaction_id] / df[cols.agg_customer_id]
+
+    p1_df = df[p1_index]
+    p1_df.columns = [col + "_" + get_option("column.suffix.period_1") for col in p1_df.columns]
+    p2_df = df[p2_index]
+    p2_df.columns = [col + "_" + get_option("column.suffix.period_2") for col in p2_df.columns]
+
+    # When df only contains two periods than the indexes should be dropped for proper concatenation
+    if len(df.index) == 2:  # noqa: PLR2004
+        p1_df = p1_df.reset_index(drop=True)
+        p2_df = p2_df.reset_index(drop=True)
+
+    # fillna with 0 to handle cases when one time period isn't present
+    df = pd.concat([p1_df, p2_df], axis=1).fillna(0)
+
+    for col in [
+        cols.agg_customer_id,
+        cols.agg_transaction_id,
+        cols.agg_unit_spend,
+        cols.calc_spend_per_trans,
+        cols.calc_trans_per_cust,
+        cols.calc_spend_per_cust,
+    ]:
+        # Difference calculations
+        df[col + "_" + get_option("column.suffix.difference")] = (
+            df[col + "_" + get_option("column.suffix.period_2")] - df[col + "_" + get_option("column.suffix.period_1")]
+        )
+
+        # Percentage change calculations
+        df[col + "_" + get_option("column.suffix.percent_difference")] = (
+            df[col + "_" + get_option("column.suffix.difference")]
+            / df[col + "_" + get_option("column.suffix.period_1")]
+        )
+
+    # Calculate price elasticity
+    if cols.agg_unit_qty in df_cols:
+        df["calc_price_elasticity"] = (
+            (
+                df[cols.agg_unit_spend + "_" + get_option("column.suffix.period_2")]
+                - df[cols.agg_unit_spend + "_" + get_option("column.suffix.period_1")]
+            )
+            / (
+                (
+                    df[cols.agg_unit_spend + "_" + get_option("column.suffix.period_2")]
+                    + df[cols.agg_unit_spend + "_" + get_option("column.suffix.period_1")]
+                )
+                / 2
+            )
+        ) / (
+            (
+                df[cols.calc_price_per_unit + "_" + get_option("column.suffix.period_2")]
+                - df[cols.calc_price_per_unit + "_" + get_option("column.suffix.period_1")]
+            )
+            / (
+                (
+                    df[cols.calc_price_per_unit + "_" + get_option("column.suffix.period_2")]
+                    + df[cols.calc_price_per_unit + "_" + get_option("column.suffix.period_1")]
+                )
+                / 2
+            )
+        )
```

Comment:
There exists _p1 and _p2 variables in the column helper. Eg see [here](https://github.com/Data-Simply/pyretailscience/pull/158/files#diff-1db0925d9eba18baca72a01c6a87b2906ce03a7fb7640127a8684eb955bfe9a5R136) a few lines below. This can then be simplified to,
```suggestion
        df["calc_price_elasticity"] = (
            (df[cols.agg_unit_spend_p2] - df[cols.agg_unit_spend_p1])
            / ((df[cols.agg_unit_spend_p2] + df[cols.agg_unit_spend_p1]) / 2)
        ) / (
           (df[cols.calc_price_per_unit_p2] - df[cols.calc_price_per_unit_p1])
            / ((df[cols.calc_price_per_unit_p2] + df[cols.calc_price_per_unit_p1]) / 2)
        )
```